src/k8s/pkg/k8sd/api/endpoints.go

```diff
@@ -2,55 +2,63 @@
 package api
 
 import (
-	"github.com/canonical/microcluster/microcluster"
 	"github.com/canonical/microcluster/rest"
 )
 
+type Endpoints struct {
+	provider Provider
+}
+
+// New creates a new Endpoints instance.
+func New(provider Provider) *Endpoints {
+	return &Endpoints{provider: provider}
+}
+
 // Endpoints returns the list of endpoints for a given microcluster app.
-func Endpoints(app *microcluster.MicroCluster) []rest.Endpoint {
+func (e *Endpoints) Endpoints() []rest.Endpoint {
```

Comment:
nit: `Endpoints.Endpoints()` reads a bit weird. Maybe call this `API`?

Reply:
Usage is `api.New(app, snap).Endpoints()`, should we maybe rename the type to `*api.API`?